lib/models/requestContext.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+const assert = require('../utils/assertType');
+
+// Private variables
+let
+  _connectionId = new WeakMap(),
+  _protocol = new WeakMap(),
+  _token = new WeakMap(),
+  _user = new WeakMap();
```

Comment:
Shouldn't these be defined in the constructor instead?
Defining these vars in the module scope make them behave like static properties to me.